pkg/cloudprovider/providers/nifcloud/nifcloud_elastic_load_balancer_test.go

```diff
@@ -0,0 +1,1473 @@
+				for i := range updatedELB {
+					updatedELB[i].VIP = testIPAddress
+					updatedELB[i].NetworkInterfaces[0].SystemIpAddresses = []string{"192.168.0.10", "192.168.0.11"}
+					updatedELB[i].BalancingTargets = append(updatedELB[i].BalancingTargets, *registeredInstance)
+				}
+				testSecurityGroups := helper.NewTestEmptySecurityGroups()
+
+				expectedStatus := &corev1.LoadBalancerStatus{
+					Ingress: []corev1.LoadBalancerIngress{
+						{
+							IP: testIPAddress,
+						},
+					},
+				}
+
+				c := nifcloud.NewMockCloudAPIClient(ctrl)
+				c.EXPECT().
+					DescribeElasticLoadBalancers(gomock.Any(), gomock.Eq(loadBalancerName)).
+					Return(existedELB, nil).
+					Times(1)
+				c.EXPECT().
+					RegisterInstancesWithElasticLoadBalancer(gomock.Any(), gomock.Eq(&existedELB[0]), gomock.Eq([]nifcloud.Instance{*registeredInstance})).
+					Return(nil).
+					Times(1)
+				expectedInstanceIDs := []string{registeredInstance.InstanceID}
+				c.EXPECT().
+					DescribeSecurityGroupsByInstanceIDs(gomock.Any(), gomock.Eq(expectedInstanceIDs)).
+					Return(testSecurityGroups, nil).
+					Times(1)
+				createdSecurityGroupRules := []nifcloud.SecurityGroupRule{
+					{
+						IpProtocol: updatedELB[0].Protocol,
+						FromPort:   updatedELB[0].InstancePort,
+						ToPort:     updatedELB[0].InstancePort,
+						InOut:      "IN",
+						IpRanges:   []string{testIPAddress},
+					},
+					{
+						IpProtocol: updatedELB[0].Protocol,
+						FromPort:   updatedELB[0].InstancePort,
+						ToPort:     updatedELB[0].InstancePort,
+						InOut:      "IN",
+						IpRanges:   []string{updatedELB[0].NetworkInterfaces[0].SystemIpAddresses[0]},
+					},
+					{
+						IpProtocol: updatedELB[0].Protocol,
+						FromPort:   updatedELB[0].InstancePort,
+						ToPort:     updatedELB[0].InstancePort,
+						InOut:      "IN",
+						IpRanges:   []string{updatedELB[0].NetworkInterfaces[0].SystemIpAddresses[1]},
+					},
+				}
+				callAuthorizeSecurityGroupIngressTime := 0
+				c.EXPECT().
+					AuthorizeSecurityGroupIngress(gomock.Any(), gomock.Eq(testSecurityGroups[0].GroupName), gomock.Any()).
+					Do(func(_ context.Context, _ string, securityGroupRule *nifcloud.SecurityGroupRule) {
+						Expect(*securityGroupRule).Should(Equal(createdSecurityGroupRules[callAuthorizeSecurityGroupIngressTime]))
+						callAuthorizeSecurityGroupIngressTime += 1
+					}).
+					Return(nil).
+					Times(3)
+				c.EXPECT().
+					WaitSecurityGroupApplied(gomock.Any(), gomock.Eq(testSecurityGroups[0].GroupName)).
+					Return(nil).
+					Times(3)
+
+				cloud := &nifcloud.Cloud{}
+				cloud.SetClint(c)
+				cloud.SetRegion(region)
+
+				status, err := nifcloud.ExportEnsureElasticLoadBalancer(cloud, ctx, loadBalancerName, testDesire)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(*status).Should(Equal(*expectedStatus))
+			})
+		})
+
+		Context("deregister an instance to the elastic load balancer", func() {
```

Comment:
deregister logics removes instances `from` LB, so

```suggestion
		Context("deregister an instance from the elastic load balancer", func() {
```